datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -1689,7 +1874,524 @@ pub fn interval_scalar_interval_op(
             scalar.get_datatype(),
         )))?,
     };
-    Ok(ColumnarValue::Array(ret))
+    Ok(ret)
+}
+
+/// This function handles interval +/- interval operations where the former is
+/// a scalar and the latter is an array, resulting in an interval array.
+pub fn scalar_interval_op_interval(
+    scalar: &ScalarValue,
+    sign: i32,
+    array: &ArrayRef,
+) -> Result<ArrayRef> {
+    let ret = match (scalar, array.data_type()) {
+        // YearMonth op YearMonth
+        (
+            ScalarValue::IntervalYearMonth(Some(lhs)),
+            DataType::Interval(IntervalUnit::YearMonth),
+        ) => {
+            let array = as_interval_ym_array(&array)?;
+            let ret: PrimitiveArray<IntervalYearMonthType> =
+                unary(array, |rhs| op_ym(*lhs, rhs, sign));
+            Arc::new(ret) as _
```

Comment:
`Arc::new(ret) as _` occurs many times, is there any way we can avoid it?

Reply:
Thank you for the review. I cannot figure out getting rid of `as _ `, but your other advice of adding namespaces has significantly decreased the number of lines.